sos/report/plugins/perccli2.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+
+
+class PercCLI2(Plugin, IndependentPlugin):
+
+    short_desc = 'PowerEdge RAID Controller 2 management'
+
+    plugin_name = 'perccli2'
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('perccli2',)
+
+    option_list = [
+        PluginOpt('json', default=False, desc='collect data in JSON format')
+    ]
+
+    def setup(self):
+        cmd = '/opt/MegaRAID/perccli/perccli2'
+        subcmds = [
+            'show ctrlcount',
+            '/call show AliLog',
+            '/call show all',
+            '/call show termlog',
+            '/call/bbu show all',
+            '/call/cv show all',
+            '/call/dall show',
+            '/call/eall show all',
+            '/call/eall/sall show all',
+            '/call/sall show all',
+            '/call/vall show all',
+        ]
+
+        json = ' J' if self.get_option('json') else ''
+
+        logpath = self.get_cmd_output_path()
+
+        for subcmd in subcmds:
+            self.add_cmd_output(
+                f"{cmd} {subcmd}{json}",
+                suggest_filename=f"perccli2_{subcmd}{json}",
```

Comment:
I'm not actually sure, this was a direct duplication of the existing perccli.py plugin which runs the perccli64 binary. Since the perccli2 has essentially the same usage menu as the original perccli64 binary the only modifications are:
```
--- perccli.py	2024-10-18 10:47:02.314963829 -0700
+++ perccli2.py	2024-10-18 10:50:05.080160652 -0700
@@ -9,20 +9,20 @@
 from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
 
 
-class PercCLI(Plugin, IndependentPlugin):
+class PercCLI2(Plugin, IndependentPlugin):
 
-    short_desc = 'PowerEdge RAID Controller management'
+    short_desc = 'PowerEdge RAID Controller 2 management'
 
-    plugin_name = 'perccli'
+    plugin_name = 'perccli2'
     profiles = ('system', 'storage', 'hardware',)
-    packages = ('perccli',)
+    packages = ('perccli2',)
 
     option_list = [
         PluginOpt('json', default=False, desc='collect data in JSON format')
     ]
 
     def setup(self):
-        cmd = '/opt/MegaRAID/perccli/perccli64'
+        cmd = '/opt/MegaRAID/perccli/perccli2'
         subcmds = [
             'show ctrlcount',
             '/call show AliLog',
@@ -44,7 +44,7 @@
         for subcmd in subcmds:
             self.add_cmd_output(
                 f"{cmd} {subcmd}{json}",
-                suggest_filename=f"perccli64_{subcmd}{json}",
+                suggest_filename=f"perccli2_{subcmd}{json}",
                 runat=logpath)
 
         # /call show events need 'file=' option to get adapter info like below
@@ -52,7 +52,7 @@
         subcmd = '/call show events'
         self.add_cmd_output(
             f"{cmd} {subcmd} file=/dev/stdout{json}",
-            suggest_filename=f"perccli64_{subcmd}{json}",
+            suggest_filename=f"perccli2_{subcmd}{json}",
             runat=logpath)
 
 # vim: set et ts=4 sw=4 :
```

So whatever behavior the original plugin exhibits this one would as well.

Reply:
I'll run the original and new plugin today and take a look at what the resulting filename looks like.

If this file does contain a space I'll fix it. In this case, would you like a separate PR to be opened to align the perccli.py plugin to the new behavior?

Thanks

Reply:
@pmoravec It appears as if the add_cmd_output converts spaces in suggested filenames to underscores the same as a command with spaces:
```
# ls sosreport-localhost-2024-10-21-mkiofny/sos_commands/perccli2/
perccli2_.call.bbu_show_all_J  perccli2_.call.eall.sall_show_all_J  perccli2_.call_show_AliLog_J  perccli2_.call_show_termlog_J
perccli2_.call.cv_show_all_J   perccli2_.call.eall_show_all_J       perccli2_.call_show_all_J     perccli2_.call.vall_show_all_J
perccli2_.call.dall_show_J     perccli2_.call.sall_show_all_J       perccli2_.call_show_events_J  perccli2_show_ctrlcount_J
```